index.js

```diff
@@ -212,7 +218,12 @@ NYC.prototype._maybeInstrumentSource = function (code, filename, relFile) {
     return null
   }
 
-  return this.transform(code, {filename: filename, relFile: relFile})
+  var ext = Object.keys(this.transform).filter(function (ext) {
+    return filename.toUpperCase().substr(-ext.length) === ext.toUpperCase()
+  })
```

Comment:
This function gets called *a lot*. Using `this.extensions` should help. Perhaps you could use a `for` loop and break once you've matched an extension, rather than filtering a full array?